lib/src/r_tree/r_tree_contributor.dart

```diff
@@ -18,17 +18,17 @@ part of r_tree;
 
 /// The base definition of an object that exists in an [RTree]
 abstract class RTreeContributor {
-  Rectangle get rect;
+  Rectangle? get rect;
 
   // Calculate if otherRect overlaps with the current rectangle
   //
   // This function is a replication of Rectangle.intersects. It differs in that
   // the inequalities are strict and do not allow for equivalences. This means
   // that the two rectangles are not considered overlapping if they share an edge.
   bool overlaps(Rectangle otherRect) {
```

Comment:
Instead of null checking every property, I'd create a zero sized rect if `rect` is null.
```suggestion
  bool overlaps(Rectangle otherRect) {
  final thisRect = rect ?? Rectangle(0, 0, 0, 0);
```